parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/RewriteOptions.java

```diff
@@ -358,6 +466,9 @@ public Builder indexCacheStrategy(IndexCache.CacheStrategy cacheStrategy) {
     public RewriteOptions build() {
       Preconditions.checkArgument(inputFiles != null && !inputFiles.isEmpty(), "Input file is required");
       Preconditions.checkArgument(outputFile != null, "Output file is required");
+      Preconditions.checkArgument(
+          inputFiles != null && !inputFiles.isEmpty(),
```

Comment:
Duplicate check?

Reply:
You mean input file duplicate? I did it as it was done for `inputFiles`, should I add such a check for both file input lists?

By the way, this check was supposed to be for `inputFilesToJoin` not `inputFiles`, I fixed it.

Reply:
I mean the L467 has the same check:
```java
Preconditions.checkArgument(inputFiles != null && !inputFiles.isEmpty(), "Input file is required");
```

Reply:
Oh, yeah it was supposed to be for `inputFilesToJoin` not `inputFiles`. I replaced that check with `(inputFilesToJoin != null ? inputFilesToJoin : new ArrayList<>())` bellow. Hope this works.